### PR TITLE
Replace using statements with using declarations in solution

### PR DESCRIPTION
**Pull request by keyboardDrummer**
This PR replaces using statements with using declarations to reduce unhelpful syntax. More information is in the release notes of the feature: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using#motivation

### CODE REVIEW

File: Source/Predication/SmartBlockPredicator.cs [hunk 1]
```diff
@@ -328,7 +328,7 @@ Variable FreshPredicate(ref int predCount)
               var condPred = FreshPredicate(ref predCount);
               predMap[block.Item1] = condPred;
               defMap[block.Item1] = condPred;
-              var headerIterator = headerDominance.GetEnumerator();
+              using var headerIterator = headerDominance.GetEnumerator();
```
Comment (shazqadeer) [inline Source/Predication/SmartBlockPredicator.cs:331]: What is the difference in behavior as a result of this change?

**Comment by shazqadeer** [comment]
What is the primary motivation for this PR?

File: Source/Predication/SmartBlockPredicator.cs (hunk 1 above)
Comment (keyboardDrummer) [reply Source/Predication/SmartBlockPredicator.cs:331]: I'm guessing no practical difference, but now the method satisfies the contract of `GetEnumerator`, which is that it should be disposed after it's used. This isn't required when using `foreach` to enumerate, and I'm not sure why foreach is not used here.

**Comment by keyboardDrummer** [comment]
> What is the primary motivation for this PR?

The motivation is to reduce the amount of syntax, the same as why using declarations were introduced into the language: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using#motivation

**Review by shazqadeer — APPROVED** [review]

**Pull request merged**